Router/DefaultPatternGenerationStrategy.php

```diff
@@ -44,14 +45,20 @@ public function generateI18nPatterns($routeName, Route $route)
     {
         $patterns = array();
         foreach ($route->getOption('i18n_locales') ?: $this->locales as $locale) {
-            // Check if translation exists in the translation catalogue to avoid errors being logged by 
+            // Check if translation exists in the translation catalogue to avoid errors being logged by
             // the new LoggingTranslator of Symfony 2.6. However, the LoggingTranslator did not implement
             // the interface until Symfony 2.6.5, so an extra check is needed.
             if ($this->translator instanceof TranslatorBagInterface || $this->translator instanceof LoggingTranslator) {
                 // Check if route is translated.
                 if (!$this->translator->getCatalogue($locale)->has($routeName, $this->translationDomain)) {
                     // No translation found.
-                    $i18nPattern = $route->getPattern();
+                    // getPattern is deprecated since Symfony 2.2 and will be removed in 3.0. Use the getPath() message suppressed.
+                    if (Kernel::VERSION > 2.2) {
```

Comment:
checking the kernel version is wrong. The routing component and the HttpKernel component could be installed at different versions.

The right way is to drop support for 2.1, which is unmaintained for years anyway